### PR TITLE
workaround for dictionary_encode with string type 

### PR DESCRIPTION
**Pull request by marc9595**
Workaround for https://issues.apache.org/jira/browse/ARROW-7266

Currently, this is happening:
```
In: aa = pa.array(['a','b','b','b'])
In: fr.FletcherArray(aa)[1:].factorize()
Out: (array([0, 0, 1]), <FletcherArray>
 ['b', '\x00']
 Length: 2, dtype: fletcher[string])
In:indices, unique = fr.FletcherArray(aa)[1:].factorize()
In:unique.take(indices)
Out: <FletcherArray>
['b', 'b', '\x00']
Length: 3, dtype: fletcher[string]
```
Apparently, this issue appears only with string type when the offset is not 0. I used the workaround given in the commentary and added some unittests.

### CODE REVIEW

File: fletcher/base.py [hunk 1]
```diff
@@ -568,7 +568,13 @@ def factorize(self, na_sentinel=-1):
         if self.data.num_chunks == 0:
             return type(self)(pa.array([], type=self.data.type)).factorize(na_sentinel)
         else:
-            encoded = self.data.dictionary_encode()
+            if pa.types.is_string(self.data.type):
+                encoded = pa.chunked_array(
+                    pa.serialize(chunk).deserialize() if chunk.offset else chunk
```
Comment (artemru) [inline fletcher/base.py]:
```suggestion
                    pa.serialize(chunk).deserialize() if chunk.offset != 0 else chunk
```

Comment (artemru) [reply fletcher/base.py]: better not to abuse python conversion to bool

File: tests/test_base.py [hunk 2]
```diff
@@ -301,3 +301,27 @@ def test_flatten():
 
     npt.assert_array_equal(fr_test.flatten(), fr.FletcherArray([1, 2, 3, 4]))
     npt.assert_array_equal(fr_test_empty_array.flatten(), np.array([], dtype=np.int64))
+
+
+def test_factorize():
```
Comment (artemru) [inline tests/test_base.py]: don't we already have `test_factoize` somewhere ?

File: fletcher/base.py [hunk 3]
```diff
@@ -568,7 +568,13 @@ def factorize(self, na_sentinel=-1):
         if self.data.num_chunks == 0:
             return type(self)(pa.array([], type=self.data.type)).factorize(na_sentinel)
         else:
-            encoded = self.data.dictionary_encode()
+            if pa.types.is_string(self.data.type):
+                encoded = pa.chunked_array(
```
Comment (artemru) [inline fletcher/base.py]: please add pyarrow bug ref here

File: tests/test_base.py (hunk 2 above)
Comment (marc9595) [reply tests/test_base.py]: Yes we do in `test_pandas_integration.py`, I just checked. I should only test `factorize` when there is an offset, then.

File: fletcher/base.py (hunk 1 above)
Comment (marc9595) [reply fletcher/base.py]: Done.

**Comment by marc9595** [comment]
Thanks for the reviews ! Merging.

**Pull request merged**